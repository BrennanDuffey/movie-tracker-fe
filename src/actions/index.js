export const addMovies = (movies) => ({
  type: "FETCH_MOVIES",
  movies
});

export const grabCurrMovie = (movie) => ({
  type: "GRAB_MOVIE",
  movie
});

export const isLoading = (isLoading) => ({
  type: "IS_LOADING",
  isLoading
});

export const errorMessage = (errorMessage) => ({
  type: "SET_ERROR_MESSAGE",
  errorMessage
});

export const toggleFavorite = (id) => ({
  type: "TOGGLE_FAVORITE",
  id
}); 
        
export const loginUser = (user) => ({
  type: "LOGIN_USER",
  user
});
        
export const signoutUser =()=>({
  type: "SIGNOUT_USER"
});

export const setFavorites = (favorites) => ({
  type: "SET_FAVORITES",
  favorites
});