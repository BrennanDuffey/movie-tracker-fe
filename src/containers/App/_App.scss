html {
  background-color: black;
}

.App {
  position: relative;
  margin: 0;
  padding: 0;
  width: 100vh;
  height: 100vh;
  display: grid;
  grid-template-columns: 20% 80%;
  justify-content: center;
}