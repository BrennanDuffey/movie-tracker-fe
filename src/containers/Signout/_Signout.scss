.Signout {
  background-color: black;
  width: 89vw;
  display: flex;
  justify-content: center;
  align-items: center;
  position: relative;
  background: url('https://images.unsplash.com/photo-1489599849927-2ee91cede3ba?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1950&q=80');
  background-size: cover;
}

.Signout-divider {
  content: '';
  height: 5px;
  width: 80%;
  background-color: #500000;
  border-radius: 30%;
}

.Signout-form {
  display: flex;
  justify-content: space-evenly;
  flex-direction: column;
  align-items: center;
  background-color: #970000;
  border-radius: 35px;
  width: 50vw;
  height: 25vw;
  box-shadow: black 5px 5px 10px;

    h2 {
      color: white;
      font-size: 35px;
      text-shadow: .5px .5px 2px black;
      margin: 0;
    }

    h4 {
      color: white;
      text-shadow: .5px .5px 1.8px black;
      font-size: 22px;
      margin: 0;
    }

    .user-submit {
      background: #500000;
      text-transform: uppercase;
      color: white;
      height: 50px;
      width: 120px;
      border-radius: 15px;

        &:hover {
          background: #3a3a3a;
          box-shadow: 2px 2px 6px white;
          cursor: pointer;
        }
    }
}
