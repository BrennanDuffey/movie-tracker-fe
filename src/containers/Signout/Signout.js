import React, {Component} from 'react';
import {connect} from 'react-redux';
import { Redirect } from "react-router-dom";
import {signoutUser} from '../../actions';

export class Signout extends Component {
  constructor(){
    super()
    this.state={
      loggedOut: false
    }
  }

  handleSubmit=(e)=>{
    e.preventDefault();
    this.setState({loggedOut: true})
    this.props.signoutUser();
  }
  render() {
    if (this.state.loggedOut) {
      return <Redirect to='/' />
    }

    return (
      <section className="Signout">
        <form className="Signout-form" onSubmit={this.handleSubmit}>
          <h2>Leaving so soon?</h2>
          <div className="Signout-divider"></div>
          <span className="h-group">
            <h4>Click the button below,</h4>
            <h4>To signout of ButterFlix</h4>
          </span> 
          <input className="user-submit" type="submit" value="Signout"/>
        </form>
      </section>
    )
  }
}

const mapDispatchToProps =(dispatch)=> ({
  signoutUser: ()=> dispatch(signoutUser())
});

export default connect(null, mapDispatchToProps)(Signout);