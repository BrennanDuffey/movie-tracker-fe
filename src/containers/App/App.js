import React, {Component} from 'react';
import {Route} from 'react-router-dom';
import { connect } from 'react-redux';
import NavBar from '../NavBar/NavBar';
import CardContainer from '../CardContainer/CardContainer';
import BigCard from "../BigCard/BigCard";
import FavoriteCardContainer from '../FavoriteCardContainer/FavoriteCardContainer';
import UserForm from '../UserForm/UserForm';
import Signout from '../Signout/Signout';
import { fetchMovies } from '../../utils/apiCalls/apiCalls';
import { addMovies } from '../../actions';
import { cleanFetchMovies } from '../../utils/cleaners/cleanMovies';

export class App extends Component {
  constructor() {
    super();
  }

  componentDidMount() {
    fetchMovies()
      .then(list => list.results)
      .then(movies=> cleanFetchMovies(movies))
      .then(cleanMovies => this.props.addMovies(cleanMovies))
      .catch(error => console.log('Error in fetching movies from DB'))
  }

  render(){
    return (
      <div className="App">
        <NavBar />
        <Route exact path='/' component={CardContainer} />
        <Route path={`/movie/`} component={BigCard} />
        <Route exact path='/login' component={UserForm}/>
        <Route exact path='/signout' component={Signout}/>
        <Route exact path='/favorites' component={FavoriteCardContainer} />
      </div>
    )
  }
}


export const mapDispatchToProps = (dispatch) => ({
  addMovies: (movies) => dispatch(addMovies(movies))
});


export default connect(null, mapDispatchToProps)(App);
