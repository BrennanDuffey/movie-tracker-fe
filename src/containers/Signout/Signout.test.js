import React from 'react';
import {Signout} from './Signout';
import {shallow} from 'enzyme';
import { signoutUser } from '../../actions';
import { mapDispatchToProps } from '../App/App';

let signoutUserMock = jest.fn();

describe('Signout', ()=>{
  let wrapper;

  beforeEach(()=>{
    wrapper = shallow(<Signout signoutUser={signoutUserMock} />);
  })

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on submittion of form', ()=>{
    expect(wrapper.state('loggedOut')).toEqual(false);
    
    wrapper.find('form').simulate('submit',
      {preventDefault: ()=>{}});

    expect(wrapper.state("loggedOut")).toEqual(true);
  });

  it('should invoke signoutUser on form submittion', ()=>{
    wrapper.find('form').simulate('submit',
      {preventDefault: ()=>{}});

    expect(signoutUserMock).toHaveBeenCalled();
  });
});