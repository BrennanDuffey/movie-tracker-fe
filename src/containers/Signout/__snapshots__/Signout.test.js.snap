// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signout should match snapshot 1`] = `
<section
  className="Signout"
>
  <form
    className="Signout-form"
    onSubmit={[Function]}
  >
    <h2>
      Leaving so soon?
    </h2>
    <div
      className="Signout-divider"
    />
    <span
      className="h-group"
    >
      <h4>
        Click the button below,
      </h4>
      <h4>
        To signout of ButterFlix
      </h4>
    </span>
    <input
      className="user-submit"
      type="submit"
      value="Signout"
    />
  </form>
</section>
`;
